src/huggingface_hub/snapshot_download.py

```diff
@@ -53,18 +60,46 @@ def snapshot_download(
     else:
         token = None
 
-    _api = HfApi()
-    model_info = _api.model_info(repo_id=repo_id, revision=revision, token=token)
+    # remove all `/` occurances to correctly convert repo to directory name
+    repo_id_flattened = repo_id.replace("/", REPO_ID_SEPARATOR)
 
-    storage_folder = os.path.join(
-        cache_dir, repo_id.replace("/", REPO_ID_SEPARATOR) + "." + model_info.sha
-    )
+    # if we have no internet connection we will look for the
+    # last modified folder in the cache
+    if local_files_only:
+        # list all possible folders that correspond to the repo_id
+        repo_id_folders = glob.glob(
+            os.path.join(cache_dir, repo_id_flattened + "." + revision + ".*")
+        )
+
+        if len(repo_id_folders) == 0:
+            raise ValueError(
+                "Cannot find the requested files in the cached path and outgoing traffic has been"
+                " disabled. To enable model look-ups and downloads online, set 'local_files_only'"
+                " to False."
+            )
+
+        # find last modified folder
+        storage_folder = max(
+            repo_id_folders,
+            key=os.path.getmtime,
+        )
 
-    for model_file in model_info.siblings:
-        url = hf_hub_url(
-            repo_id, filename=model_file.rfilename, revision=model_info.sha
+        repo_id_sha = storage_folder.split(".")[-1]
+        model_files = os.listdir(storage_folder)
+    else:
+        # if we have internet connection we retrieve the correct folder name from the huggingface api
+        _api = HfApi()
+        model_info = _api.model_info(repo_id=repo_id, revision=revision, token=token)
+        storage_folder = os.path.join(
+            cache_dir, repo_id_flattened + "." + revision + "." + model_info.sha
```

Comment:
If the `revision` and `model_info.sha` are identical (which they are if you explicitly specify the `revision` parameter to a commit hash), then this will put it twice in the name. Is this expected? i.e., doing:

```py
snapshot_download("xxx", revision="yyy")
# ~/.cache/xxx.yyy.yyy
```

Reply:
(not necessarily a problem)